src/Core/CFamily/CFamilyBindingConfigProvider.cs

```diff
@@ -68,7 +75,12 @@ public async Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProf
             cancellationToken.ThrowIfCancellationRequested();
 
             var settings = CreateRulesSettingsFromQPRules(result);
-            var configFile = new CFamilyBindingConfigFile(settings);
+            var settingsFilePath = solutionBindingFilePathGenerator.Generate(
```

Comment:
This is where the magic happens -- calculating and assigning the path of the ruleset to IBindingConfig

Reply:
I'm not sure the `ISolutionBindingFilePathGenerator` interface will survive much longer; I'm not convinced it's still necessary or appropriate.

Reply:
What do you think we should do instead?

Reply:
The interface centralises the file-naming logic for all languages.
In general, the refactoring is devolving language-specific handling to language-specific classes. I'm questioning whether the centralised interface is still appropriate, particularly since the additional file naming logic will be specific to C#/VB.

Either there should be a language-specific name calculator, or the binder should just handle the calculation itself. I don't think it need to be centralised, and I'm not sure it needs to be pluggable for testing (for C++, the logic is simple enough to be handled in a static method on the CFamily binder).

Have a look at `Language.FileSuffixAndExtension`. I had to add it to make the file name calculation work when I added support for CFamily connected mode. I don't think that property should be needed.

Either way, it doesn't affect this PR.